internal/search/query/transformer.go

```diff
@@ -563,3 +563,67 @@ func FuzzifyRegexPatterns(nodes []Node) []Node {
 		return Parameter{Field: field, Value: value, Negated: negated, Annotation: annotation}
 	})
 }
+
+// mapRevFilters removes rev: filters from []Node and attaches their value as @rev to the repo: filters.
+// All rev: filters are assumed to be on the top-level, i.e. mapRevFilter will not traverse the tree.
```

Comment:
```suggestion
// Invariant: we assume that all rev: filters are at the top-level of []Node, or in one of it's direct children. This invariant is ensured when the query is normalized to DNF.
```

I've thought about it and mostly convinced myself that _if_ the query is in normal form (i.e., all `or`-expressions have been distributed) then it is safe to assume that all `rev:`s will always be at the top level (either in `[]Node` or in one of it's direct children). If this were not the case I would have suggested we change the code to perform this operation recursively.

I'm not a huge fan of commenting about invariants like this when we have this wonderful way to encode invariants/constraints in something called types :P So I may introduce a type for normalized queries to guarantee this property. For now this is fine.

Reply:
My reasoning was that right now `rev:` will be at the top level because we don't support expressions for filters yet. Once we support expressions, we will normalize the query and `rev:` will again be at the top level. So all tree traversals turn into simple loops anyways.

> So I may introduce a type for normalized queries to guarantee this property

I like the idea 👍

Reply:
> My reasoning was that right now rev: will be at the top level because we don't support expressions for filters yet.

Yep we're on the same page.